ethereum/event/BREvent.c

```diff
@@ -268,10 +299,17 @@ eventHandlerStop (BREventHandler handler) {
         // Quit the thread by aborting the queue wait.
         eventQueueDequeueWaitAbort (handler->queue);
 
-        // Wait for the thread.
-        pthread_join (handler->thread, NULL);
-        // A mini-race here?
-        handler->thread = PTHREAD_NULL;
+        // Wait for the thread.  We must release the lock here because the `eventHandlerThread`
+        // could itself be waiting on the lock (only as part of a running, dispatched function).
+        // Once we give the lock, the dispatched function will complete, the `eventHandlerThread`
+        // will return to the EventQueue and see the above 'Wait Abort' cond-var signaled.  At
+        // that point, `eventHandlerThread` will finish and signal 'threadExit`.  The following
+        // cond_wait will complete and reaquire the lock.
+        //
+        // Note, while we wait we will not hold the lock but handle->thread will be non-NULL; so,
+        // we don't have a start/stop race.
+        //
+        pthread_cond_wait (&handler->threadExit, &handler->lock);
```

Comment:
Think a `pthread_join` is required to clean up thread resources, no?

Reply:
Yes